File: Src2/Icm.Core/Repository/MemoryEntityRepositoryOfTEntityTKey.vb
Namespace Icm.Data

    Public Class MemoryEntityRepository(Of TEntity As IEntity(Of TKey), TKey)
        Implements IRepository(Of TEntity, TKey)

        Protected store As Dictionary(Of TKey, TEntity)

        ''' <summary>
        ''' Initializes a new instance of the MemoryRepository class.
        ''' </summary>
        Public Sub New()
            store = New Dictionary(Of TKey, TEntity)
        End Sub

        ''' <summary>
        ''' Initializes a new instance of the MemoryRepository class.
        ''' </summary>
        ''' <param name="items"></param>
        Public Sub New(ByVal items As ICollection(Of TEntity))
            Load(items)
        End Sub

        Public Sub Load(items As IEnumerable(Of TEntity))
            Me.store = New Dictionary(Of TKey, TEntity)
            For Each item In items
                Add(item)
            Next
        End Sub

        Public Sub Add(entity As TEntity) Implements IRepository(Of TEntity, TKey).Add
            store.Add(entity.Id, entity)
        End Sub

        Public Sub Delete(entity As TEntity) Implements IRepository(Of TEntity, TKey).Delete
            store.Remove(entity.Id)
        End Sub

        Public Overridable Function GetById(id As TKey) As TEntity Implements IRepository(Of TEntity, TKey).GetById
            Return store(id)
        End Function

        Public Function GetEnumerator() As IEnumerator(Of TEntity) Implements IEnumerable(Of TEntity).GetEnumerator
            Return store.Values.GetEnumerator
        End Function

        Public Function GetEnumerator1() As IEnumerator Implements IEnumerable.GetEnumerator
            Return store.Values.GetEnumerator
        End Function

        Public Function Contains(entity As TEntity) As Boolean Implements IRepository(Of TEntity, TKey).Contains
            Return store.ContainsKey(entity.Id)
        End Function

        Public Function ContainsId(key As TKey) As Boolean Implements IRepository(Of TEntity, TKey).ContainsId
            Return store.ContainsKey(key)
        End Function
    End Class

End Namespace

File: Src/Icm.Core/Repository/MemoryEntityRepositoryOfTEntityTKey.cs
using System.Collections;
using System.Collections.Generic;

namespace Icm.Data
{

	public class MemoryEntityRepository<TEntity, TKey> : IRepository<TEntity, TKey> where TEntity : IEntity<TKey>
	{


		protected Dictionary<TKey, TEntity> store;
		/// <summary>
		/// Initializes a new instance of the MemoryRepository class.
		/// </summary>
		public MemoryEntityRepository()
		{
			store = new Dictionary<TKey, TEntity>();
		}

		/// <summary>
		/// Initializes a new instance of the MemoryRepository class.
		/// </summary>
		/// <param name="items"></param>
		public MemoryEntityRepository(ICollection<TEntity> items)
		{
			Load(items);
		}

		public void Load(IEnumerable<TEntity> items)
		{
			this.store = new Dictionary<TKey, TEntity>();
			foreach (void item_loopVariable in items) {
				item = item_loopVariable;
				Add(item);
			}
		}

		public void Add(TEntity entity)
		{
			store.Add(entity.Id, entity);
		}

		public void Delete(TEntity entity)
		{
			store.Remove(entity.Id);
		}

		public virtual TEntity GetById(TKey id)
		{
			return store(id);
		}

		public IEnumerator<TEntity> GetEnumerator()
		{
			return store.Values.GetEnumerator;
		}

		public IEnumerator GetEnumerator1()
		{
			return store.Values.GetEnumerator;
		}
		IEnumerator IEnumerable.GetEnumerator()
		{
			return GetEnumerator1();
		}

		public bool Contains(TEntity entity)
		{
			return store.ContainsKey(entity.Id);
		}

		public bool ContainsId(TKey key)
		{
			return store.ContainsKey(key);
		}
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
